avocado/utils/vmimage.py

```diff
@@ -386,9 +386,9 @@ def list_providers():
     List the available Image Providers
     """
     return set(_ for _ in itervalues(globals())
-               if (_ != ImageProviderBase and
-                   isinstance(_, type) and
-                   issubclass(_, ImageProviderBase)))
+               if (isinstance(_, type) and
+                   issubclass(_, ImageProviderBase) and
+                   hasattr(_, 'name')))
```

Comment:
I'd not mind being explicit and just return a list of image providers...
```python
def list_providers():
    """
    List the available Image Providers
    """
    return [DebianImageProvider,
            UbuntuImageProvider,
             ...]
```
I don't think we introduce so many new provides to the point we fail to keep this list up-to-date.

Reply:
Or, even better, just drop this function and create `IMAGE_PROVIDERS` directly with that list.